lib/rucio/tests/test_impl_upload_download.py

```diff
@@ -0,0 +1,528 @@
+        impl = 'xrootd'
+
+        # Adding files to a new dataset
+        cmd = 'rucio upload --rse {0} --scope {1} --impl {2} {3} {4} {5} {1}:{6}'.format(self.rse, self.scope, impl, tmp_file1, tmp_file2, tmp_file3, tmp_dsn)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out)
+        print(err)
+        remove(tmp_file1)
+        remove(tmp_file2)
+        remove(tmp_file3)
+        self.assertEqual(exitcode, 0)
+
+        # List the files
+        cmd = 'rucio list-files {0}:{1}'.format(self.scope, tmp_dsn)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out)
+        print(err)
+        self.assertEqual(exitcode, 0)
+
+        # List the replicas
+        cmd = 'rucio list-file-replicas {0}:{1}'.format(self.scope, tmp_dsn)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out)
+        print(err)
+        self.assertEqual(exitcode, 0)
+
+        # Downloading dataset
+        cmd = 'rucio download --dir /tmp/ {0}:{1} --impl {2}'.format(self.scope, tmp_dsn, impl)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out)
+        print(err)
+        # The files should be there
+        cmd = 'ls /tmp/{0}/rucio_testfile_*'.format(tmp_dsn)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(err, out)
+        self.assertEqual(exitcode, 0)
+
+        # cleaning
+        remove('/tmp/{0}/'.format(tmp_dsn) + tmp_file1[5:])
+        remove('/tmp/{0}/'.format(tmp_dsn) + tmp_file2[5:])
+        remove('/tmp/{0}/'.format(tmp_dsn) + tmp_file3[5:])
+        added_dids = ['{0}:{1}'.format(self.scope, did) for did in (tmp_file1, tmp_file2, tmp_file3, tmp_dsn)]
+        self.generated_dids += added_dids
+
+    def test_repeat_upload_file_using_impl(self):
+        """CLIENT(USER): rucio re-upload file after deleting physical file"""
+        tmp_file1 = file_generator()
+        tmp_file2 = file_generator()
+        tmp_file3 = file_generator()
+        tmp_file1_name = path.basename(tmp_file1)
+        impl = 'xrootd'
+
+        # Upload the file
+        cmd = 'rucio upload --rse {0} --scope {1} --impl {2} {3}'.format(self.rse, self.scope, impl, tmp_file1)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out)
+        print(err)
+        # get the rule for the file
+        cmd = r"rucio list-rules {0}:{1} | grep {0}:{1} | cut -f1 -d\ ".format(self.scope, tmp_file1_name)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out, err)
+        rule = out
+        # delete the file from the catalog
+        cmd = "rucio delete-rule {0}".format(rule)
+        print(self.marker + cmd)
+        exitcode, out, err = execute(cmd)
+        print(out, err)
+        # delete the physical file
+        cmd = "ls /tmp/rucio_rse/test"
```

Comment:
this actually is the statement: cmd = "find /tmp/rucio_rse/ -name {0} |xargs rm".format(tmp_file1_name)
this line change was just for me to see what was happening